Floofbot/Services/RaidProtectionService.cs

```diff
@@ -0,0 +1,387 @@
+﻿using Discord;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Eventing.Reader;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Security.Policy;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Floofbot.Services
+{
+    public class RaidProtectionService
+    {
+        // will store user id and how many counts they've had
+        private Dictionary<ulong, int> userPunishmentCount = new Dictionary<ulong, int>();
```

Comment:
`userPunishmentCount`, `userMessageCount` and `punishedUsers` are all not tied to any one guild. Thus, a user can be preemptively "primed" for a strike on a server they weren't active on. Behavior in `CheckMessage` will need to be changed as a result.

Reply:
Changed to make them guild dependent